backend/app/uploads.py

```
@@ -130,13 +131,13 @@
         return PresignedUrlResponse(
-            url=response["url"],
-            fields=response["fields"],
+            url=policy['url'],
+            fields=policy['fields'],
             fileId=file_id
         )
         
-    except ClientError as e:
-        raise HTTPException(status_code=500, detail=f"S3 error: {str(e)}")
+    except gcp_exceptions.GoogleAPICallError as e:
+        raise HTTPException(status_code=500, detail=f"GCS error: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)